systemd_ctypes/bus.py

```diff
@@ -320,124 +327,467 @@ def add_match(self, rule, handler):
+            if obj is None:
+                return self
+            if self._getter is not None:
+                return self._getter.__get__(obj, cls)()
+            elif self._value is not None:
+                if obj._dbus_property_values is not None:
+                    return obj._dbus_property_values.get(self._name, self._value)
+                else:
+                    return self._value
+            else:
+                raise AttributeError(f"'{obj.__class__.__name__}' property '{self._python_name}' "
+                                     f"was not properly initialised: use either the 'value=' kwarg or "
+                                     f"the @'{self._python_name}.getter' decorator")
+
+        def __set__(self, obj: Any, value: Any) -> None:
+            if self._getter is not None:
+                raise AttributeError(f"'{obj.__class__.__name__}' property '{self._python_name}' is read-only")
+            if obj._dbus_property_values is None:
+                obj._dbus_property_values = {}
+            obj._dbus_property_values[self._name] = value
+            obj.properties_changed(self._interface or obj._dbus_interface,
+                                   {self._name: {"t": self._type_string, "v": value}},
+                                   [])
+
+        def to_dbus(self, obj):
+            return {"t": self._type_string, "v": self.__get__(obj)}
+
+        def from_dbus(self, obj, value):
+            if self._setter is None or self._type_string != value["t"]:
+                raise Object.Method.Unhandled
+            self._setter.__get__(obj)(value["v"])
+
+        def getter(self, getter: Callable[[Any], Any]) -> Callable[[Any], Any]:
+            if self._value is not None:
+                raise ValueError('A property cannot have both a value and a getter')
+            if self._getter is not None:
+                raise ValueError('A property cannot have both a value and a getter')
+            self._getter = getter
+            return getter
+
+        def setter(self, setter: Callable[[Any, Any], None]) -> Callable[[Any, Any], None]:
+            self._setter = setter
+            return setter
+
+
+    class Signal(_Member):
+        """Defines a D-Bus signal on an interface
+
+        This is a callable which will result in the signal being emitted.
+
+        The constructor takes the types of the arguments, each one as a
+        separate parameter.  For example:
+
+            properties_changed = Interface.Signal('s', 'a{sv}', 'as')
+
+        You can give a name= kwarg to override the default name conversion
+        scheme.
+        """
+        _category = 'signals'
+
+        def __init__(self, *out_types: str, name: Optional[str] = None) -> None:
+            super().__init__(name=name)
+            self._out_types = list(out_types)
+            self._signature = ''.join(out_types)
+
+        def _describe(self) -> Dict[str, Any]:
+            return {'in': self._out_types}
```

Comment:
It's weird to mix `in` with `out_types` here, but I guess one can argue that the signal parameters should really be called `out` in the Introspection XML, but alas they aren't.

Reply:
I agree!

It's sort of a matter of perspective, though.  For the person emitting the signal, it's out, and for the person receiving it, it's in.  I think this isn't that odd.